osu.Game/Overlays/BeatmapListing/BeatmapSearchFilterRow.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.BeatmapListing
+{
+    public class BeatmapSearchFilterRow<T> : CompositeDrawable, IHasCurrentValue<T>
+    {
+        private readonly BindableWithCurrent<T> current = new BindableWithCurrent<T>();
+
+        public Bindable<T> Current
+        {
+            get => current.Current;
+            set => current.Current = value;
+        }
+
+        public BeatmapSearchFilterRow(string headerName)
+        {
+            AutoSizeAxes = Axes.Y;
+            RelativeSizeAxes = Axes.X;
+            AddInternal(new FillFlowContainer
+            {
+                AutoSizeAxes = Axes.Y,
+                RelativeSizeAxes = Axes.X,
+                Spacing = new Vector2(0, 7),
```

Comment:
If we're going to end up supporting that mode, should be

```suggestion
                Spacing = new Vector2(0, 5),
```

![image](https://user-images.githubusercontent.com/20418176/74677896-cf7b8600-51b9-11ea-8081-316a363c718f.png)

Reply:
Hmm, I made it 7 because the header has 5px bottom margin and the filter has 2px top margin
![Screenshot_1](https://user-images.githubusercontent.com/22874522/74678243-7fa5ba80-51cb-11ea-869d-568cf33b24fb.png)
![Screenshot_2](https://user-images.githubusercontent.com/22874522/74678244-803e5100-51cb-11ea-889d-3c8a7042aef6.png)

Reply:
As far as I can tell in devtools those margins shouldn't be added up as they both apply from the text baseline downwards, so the 5 actually overrides the 2.

Reply:
![Screenshot_1](https://user-images.githubusercontent.com/22874522/74679027-b250b280-51cd-11ea-9573-d666e1cc95b7.png)
Looks like they don't